completion/available/ssh.completion.bash

```diff
@@ -11,11 +11,16 @@ _sshcomplete() {
       local OPTIONS=" -- ${CURRENT_PROMPT}"
     fi
 
-
-    # parse all defined hosts from .ssh/config
-    if [ -r "$HOME/.ssh/config" ]; then
-        COMPREPLY=($(compgen -W "$(grep ^Host "$HOME/.ssh/config" | awk '{for (i=2; i<=NF; i++) print $i}' )" ${OPTIONS}) )
-    fi
+    # Parse defined hosts from all ssh configuration files (taking "Include" directives into consideration).
+    # The solution relies on ssh printing the configuration files paths in verbose mode.
+    #
+    # We try to connect to localhost on port 4, which is reserved and should be unused.
+    # In case there is an ssh server listening on that port, we ask ssh to "do nothing" by executing "echo" remotely.
+    config_files="$(ssh -v 127.0.0.1 -p 4 echo 2>&1 | sed -n '/Reading configuration data/ s/.*Reading configuration data \([^\r]*\).*/\1/gp')"
```

Comment:
This does not seem to work for me... The `\r` part stops the parsing when finding the letter `r`.

This is on macOS with Bash v3.2.57:

```bash
❯ ssh -v 127.0.0.1 -p 4 echo
OpenSSH_7.4p1, LibreSSL 2.5.0
debug1: Reading configuration data /Users/foo/.ssh/config
debug1: /Users/foo/.ssh/config line 1: Applying options for *
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: Connecting to 127.0.0.1 [127.0.0.1] port 4.
debug1: connect to address 127.0.0.1 port 4: Connection refused
ssh: connect to host 127.0.0.1 port 4: Connection refused
```

And then with the `sed` command added:

```bash
❯ ssh -v 127.0.0.1 -p 4 echo 2>&1 | sed -n '/Reading configuration data/ s/.*Reading configuration data \([^\r]*\).*/\1/gp'
/Use
/etc/ssh/ssh_config
```

As you can see the `r` in `/Users` terminates the parsing - I assume you wanted to use this to detect the end of the line...